python/mlcroissant/mlcroissant/_src/operation_graph/operations/parse_json.py

```diff
@@ -1,20 +1,219 @@
 """Parse JSON operation."""
 
+import json
+import jmespath
 import jsonpath_rw
+import orjson
 import pandas as pd
+from typing import TextIO, Any
 
-from mlcroissant._src.core.types import Json
 from mlcroissant._src.structure_graph.nodes.field import Field
+from mlcroissant._src.structure_graph.nodes.source import FileProperty
 
 
-def parse_json_content(json: Json, fields: tuple[Field, ...]) -> pd.DataFrame:
+def parse_json_content(json_obj, fields):
     """Parsed all JSONs defined in the fields of RecordSet and outputs a pd.DF."""
     series = {}
     for field in fields:
-        json_path = field.source.extract.json_path
-        if json_path is None:
+        jp = field.source.extract.json_path
+        if not jp:
             continue
-        jsonpath_expression = jsonpath_rw.parse(json_path)
-        values = [match.value for match in jsonpath_expression.find(json)]
-        series[json_path] = values
+        expr = jsonpath_rw.parse(jp)
+        vals = []
+        for match in expr.find(json_obj):
+            v = match.value
+            # if we got back a one‐item list, unwrap it
+            if isinstance(v, list) and len(v) == 1:
+                v = v[0]
+            vals.append(v)
+        series[jp] = vals
     return pd.DataFrame(series)
+
+
+class JsonReader:
```

Comment:
Any chance we can add some tests to python/mlcroissant/mlcroissant/_src/operation_graph/operations/parse_json_test.py for these two classes?